frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/task-sidebar/TaskSidebarName.tsx

```diff
@@ -40,25 +25,34 @@ const TaskSidebarName: React.FC<TaskSidebarNameProps> = (props) => {
       validated={isValid ? 'default' : 'error'}
       isRequired
     >
-      <SidebarInputWrapper>
-        <TextInput
-          id="task-name"
-          validated={isValid ? 'default' : 'error'}
-          isRequired
-          onChange={(value) => {
-            setInterimName(value);
-            setError(getError(value, t));
-          }}
-          onBlur={() => {
-            if (isValid) {
-              onChange(interimName);
-            }
-          }}
-          placeholder={taskName}
-          type={TextInputTypes.text}
-          value={interimName}
-        />
-      </SidebarInputWrapper>
+      <TextInput
+        id="task-name"
+        validated={isValid ? 'default' : 'error'}
+        isRequired
+        onChange={(value) => {
+          setInterimName(value);
+
+          setValidating(true);
+          nameValidationSchema(t, 63)
```

Comment:
If the modified name is already taken by one of the tasks or finallyTasks, then we should throw some error to force the user to provide a unique name.

Reply:
Name duplication messes up the graph structure because there are two nodes with same name.

https://user-images.githubusercontent.com/9964343/112825107-fbe61f00-90a8-11eb-8084-dc8aa4d62a3e.mp4

Reply:
Yup, good catch... this was always an issue and it was just not uncovered haha. Will fix it as part of this PR.